File: test/resources/repos/cpp/test_repo/MultipleInterfaces.cpp
#include "MultipleInterfaces.h"
#include <iostream>

MultipleInterfaces::MultipleInterfaces(const std::string& data) : data(data) {}

std::string MultipleInterfaces::read() const {
    return data;
}

void MultipleInterfaces::write(const std::string& data) {
    this->data = data;
}

void MultipleInterfaces::execute() {
    std::cout << "Executing with data: " << data << std::endl;
}

std::string MultipleInterfaces::getType() const {
    return "MultipleInterfaces";
}

File: test/resources/repos/php/test_repo/MultipleInterfaces.php
<?php

require_once 'BaseClass.php';

class MultipleInterfaces implements Readable, Writable, Processable 
{
    private string $data;

    public function __construct(string $data) 
    {
        $this->data = $data;
    }

    public function read(): string 
    {
        return $this->data;
    }

    public function write(string $data): void 
    {
        $this->data = $data;
    }

    public function process(): void 
    {
        echo "Processing data: {$this->data}\n";
    }

    public function getType(): string 
    {
        return "MultipleInterfaces";
    }
}